### PR TITLE
Feature/np cumsum

### PR DESCRIPTION
**Pull request by Plankton555**
Simple implementation of cumulative sum. Either call as a member function from an NDArray or as a static function via `np.cumsum()`.

### CODE REVIEW

File: src/NumSharp.Core/Math/NDArray.cumsum.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    {
+        public NDArray cumsum()
+        {
+            return np.cumsum(this);
+        }
+    }
+
+    public static partial class np
+    {
+        public static NDArray cumsum(NDArray a)
```
Comment (Nucs) [inline src/NumSharp.Core/Math/NDArray.cumsum.cs]: Add documentation similar to [this](https://github.com/SciSharp/NumSharp/blob/master/src/NumSharp.Core/Random/np.random.randint.cs#L14)

**Comment by Plankton555** [comment]
@Oceania2018 I tried using Linq instead but didn't get it to work. Any advice are welcome

`Assert.IsTrue(Enumerable.SequenceEqual(arr, expected));`
resulted in
`The type arguments for method 'Enumerable.SequenceEqual<TSource>(IEnumerable<TSource>, IEnumerable<TSource>)' cannot be inferred from the usage. Try specifying the type arguments explicitly`

`Assert.IsTrue(Enumerable.SequenceEqual<double>(arr, expected));`
resulted in
`cannot convert from 'NumSharp.NDArray' to 'System.Collections.Generic.IEnumerable<double>`

**Comment by Oceania2018** [comment]
`Assert.IsTrue(Enumerable.SequenceEqual(arr.GetData<double>(), expected.GetData<double>()))`

**Comment by Plankton555** [comment]
* Tests from the examples in the numpy documentation have been added.
* Added documentation to functions.

4/6 tests are failing currently. The failing tests are related to summing over specific axes, as well as specifying data types.

**Comment by Nucs** [comment]
Thank you for the PR!

**Pull request merged**